test/test_linalg.py

```diff
@@ -121,6 +121,157 @@ def run_test_case(a, b):
         run_test_case(zero_strided, b)
         run_test_case(a, zero_strided)
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_linalg_lstsq(self, device, dtype):
+        from torch.testing._internal.common_utils import random_well_conditioned_matrix
+        drivers = ('gels', 'gelsy', 'gelsd', 'gelss', None) \
```

Comment:
Style nit:

```
if self.device_type == 'cpu':
  drivers = (...)
else:
  drivers = (...)
```

is probably more readable, and encourages checking the device type vs. the device string.